cedar-policy-core/src/ast/policy.rs

```diff
@@ -266,15 +299,18 @@ impl Template {
     pub fn check_binding(
         template: &Template,
         values: &HashMap<SlotId, EntityUID>,
+        generalized_values: &HashMap<SlotId, RestrictedExpr>,
```

Comment:
I could imagine callers might not always know when to use `values` vs `generalized_values`.  Do they use `generalized_values` for all slots other than `principal` and `resource`?  or only for slots whose type isn't `EntityUID`?  It would be good to at least have a comment about this.  But I wonder if we couldn't just have all values, including EntityUIDs, be passed as a single map from `SlotId` to `RestrictedExpr`

Reply:
Based on out-of-band discussion, my understanding is we'll keep the distinction between `values` and `generalized_values` in our internal structures, so that we can have the more precise value type of `EntityUID` for `values`.  Thanks for adding the comment to this function.